### PR TITLE
Simplified ReactionSystem constructor

### PR DESCRIPTION
**Pull request by isaacsas**
Closes https://github.com/SciML/Catalyst.jl/issues/425.
CC @anandijain

### CODE REVIEW

**Comment by isaacsas** [comment]
@anandijain I did need to require passing the `iv` too as I didn't want to try to have to make a robust way to detect it...

**Comment by anandijain** [comment]
awesome! thanks a bunch

out of curiosity, what's a practical case where `t` is not the `iv`?

**Comment by isaacsas** [comment]
Don't know, but also don't want to assume... It's better to make users be explicit on what they used in constructing everything.

**Comment by isaacsas** [comment]
I have to read up a bit on how symbolics arrays work, they're breaking my conversions from `Num`s...

File: src/reactionsystem.jl [hunk 1]
```diff
@@ -207,17 +207,42 @@ function ReactionSystem(eqs, iv, species, ps;
...
+            if MT.isparameter(var) 
+                push!(ps, var)
+            else
+                push!(sts, var)
+            end
+        end
+        empty!(vars)
+    end
+
+    ReactionSystem(rxs, value(iv), collect(sts), collect(ps); skipvalue=true, kwargs...)
```
Comment (isaacsas) [inline src/reactionsystem.jl:242]:
```suggestion
    ReactionSystem(rxs, t, collect(sts), collect(ps); skipvalue=true, kwargs...)

```

File: src/reactionsystem.jl [hunk 2]
```diff
@@ -207,17 +207,43 @@ function ReactionSystem(eqs, iv, species, ps;
...
+                push!(ps, var)
+            else
+                push!(sts, var)
+            end
+        end
+        empty!(vars)
+    end
+
+    ReactionSystem(rxs, t, collect(sts), collect(ps); skipvalue=true, kwargs...)
+
```
Comment (isaacsas) [inline src/reactionsystem.jl:243]:
```suggestion
```

**Pull request merged**